src/overlay/Floodgate.cpp

```diff
@@ -120,19 +124,32 @@ Floodgate::broadcast(StellarMessage const& msg, bool force)
         releaseAssert(peer.second->isAuthenticated());
         if (peersTold.insert(peer.second->toString()).second)
         {
-            mSendFromBroadcast.Mark();
-            std::weak_ptr<Peer> weak(
-                std::static_pointer_cast<Peer>(peer.second));
-            mApp.postOnMainThread(
-                [smsg, weak, log = !broadcasted]() {
-                    auto strong = weak.lock();
-                    if (strong)
-                    {
-                        strong->sendMessage(smsg, log);
-                    }
-                },
-                fmt::format(FMT_STRING("broadcast to {}"),
-                            peer.second->toString()));
+            if (msg.type() == TRANSACTION && peer.second->shouldFloodLazily())
```

Comment:
this function needs to be reworked a little bit, compare:
* without pull mode, we send a tx to all peers that didn't send us that transaction. If we send to one peer (because we didn't receive that tx from them), the function returns `true` (`broadcasted = true`)
* with pull mode, you're doing the same thing even though you may have received an advert from a peer that they already have that transaction.
    * So this is wasting adverts *and* (because it returns `true`) causes the node to broadcast less adverts (because `TransactionQueue::broadcastSome` is rate limiting).

Reply:
I created an issue to track this: https://github.com/stellar/stellar-core/issues/3515